contracts/lib/Structs.sol

```diff
@@ -28,9 +28,10 @@ library Structs {
     }
 
     struct Lock {
-        uint256 amount; //amount in RZR
-        uint256 commission;
-        uint256 withdrawAfter; // Can be made uint32 later if packing is possible
+        uint256 amount; //amount in sRZR/RZR
```

Comment:
Yep as you mentioned I think we can do packing here

Reply:
Razor supply * decimals : 10 ** 9 * 10 ** 18 = 10 ** 27
Thats still lot less than 2 ^ 256
![image](https://user-images.githubusercontent.com/46760063/150747848-a7f0b79d-ca9d-42d0-ba42-0e8c15d704fe.png)

Reply:
I think we can have 
uint128 amount 
uint128 initial

Reply:
in that case shouldnt we do that everywhere else? uint256 is being used predominantly for stake amounts

Reply:
On commission I dont think we need to maintain so, we can leverage initial field only 
as of we have this 
```
amount
commission.  0(UnstakeLock), X(WithdrawLock)
unlockAfter
Initial              X(UnstakeLock), 0(WithdrawLock)
```
In any of locks if we have this one field is redundant 
what if we remove commssion entirely, and make it to have one lock, rather than two, hence we would save redundant ``SSTORES``

Reply:
Something like this 
Unstake 
```
Create lock 
amount : sAmount
unLockAfter : unstakeLockPeriod
Initial : razorDepostied 
```
InitiateWithdraw 
check if unlockAfter is passed

```
Update lock (not creating new)
amount : rAmount
unlockAftter: withdrawLockPeriod
initial : razorDeposted (old value only) 
```

UnlockWithdraw 
```
Calculate commission here from amount - initial
and resetLock 
(In this case only one would need to be reset)
```

If we do this we will get rid of LockType and 0 values getting stored, and storing two locks onchain 
**Let me know what do you think about it** 
Sorry I didnt spot this in previous approved review

Reply:
> in that case shouldnt we do that everywhere else? uint256 is being used predominantly for stake amounts

hmm on this yes we should if its allowing packing

Reply:
hmm I think lets put packing on hold 
We dont want it to overflow like it did that day with influence on multiplication to find commission
Please check another suggestion of above of making one lock only and removing commission from the lock as now we have initial anyway. 
CC : @SamAg19

Reply:
> hmm I think lets put packing on hold We dont want it to overflow like it did that day with influence on multiplication to find commission Please check another suggestion of above of making one lock only and removing commission from the lock as now we have initial anyway. CC : @SamAg19

Issue with the having one lock would be that how would extension work? how would we able to differentiate whether it is sRazor or Razor amount? both of them would have different mechanisms of giving extension penalty.

Reply:
yep I think we can remove extension entirely, as now we are locking razors, 😄 
lets discuss more on call

Reply:
TLDR : Removing commission from lock, and use initial